Client/Frontend/Browser/BackgroundMediaPlayback.swift

```diff
@@ -0,0 +1,110 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import AVFoundation
+import Shared
+import BraveShared
+import WebKit
+
+private let log = Logger.browserLogger
+
+class BackgroundMediaHandler {
+    private var canCleanup = false
+    private var isActive = false
+    private var shutdownRequested = false
+    private weak var profile: Profile?
```

Comment:
Why do we need to pass `Profile` here? From what I see it's only used for shutdowns

Reply:
It's needed because this class handles delayed shutdown when audio is playing. Without it, when we try to `reportSiteVisited`, it will crash due to the `db` being closed. Instead, it checks if the app or audio is active and if it is, it won't shutdown until it becomes inactive or audio stops playing.

Reply:
Can we do it in way such that the handler need not keep reference/ Actually call shutdown?